com/laboon/World.java

```diff
@@ -68,28 +71,41 @@ private State generateInitialState(int percent) {
 	 * @return
 	 */
 	
-	private int getNumNeighbors(Cell[][] world, int x, int y) {
-		int size = world.length;
-		int leftX = (x - 1) % size;
-		int rightX = (x + 1) % size;
-		int upY = (y - 1) % size;
-		int downY = (y + 1) % size;
-		
-		if (leftX == -1) { leftX = size - 1; }
-		if (rightX == -1) { rightX = size - 1; }
-		if (upY == -1) { upY = size - 1; }
-		if (downY == -1) { downY = size - 1; }
-		
+	public int getNumNeighbors(Cell[][] world, int x, int y) {
 		int numNeighbors = 0;
 
-		if (world[leftX][upY].isAlive())    { numNeighbors++; }
-		if (world[leftX][downY].isAlive())  { numNeighbors++; }
-		if (world[leftX][y].isAlive())      { numNeighbors++; }
-		if (world[rightX][upY].isAlive())   { numNeighbors++; }
-		if (world[rightX][downY].isAlive()) { numNeighbors++; }
-		if (world[rightX][y].isAlive())     { numNeighbors++; }
-		if (world[x][upY].isAlive())        { numNeighbors++; }
-		if (world[x][downY].isAlive())      { numNeighbors++; }
+		if (x != 0 && y != 0 && x != _size-1 && y != _size-1) {
+			//anywhere in the middle of the world
+			for (int i=0; i<dx.length; i++)
+				if (world[x+dx[i]][y+dy[i]].isAlive())
+					numNeighbors++;
+		} else if( x == 0 && y != 0) {
+			if (world[x][y+1].isAlive()) { numNeighbors++; }
+			if (world[x][y-1].isAlive()) { numNeighbors++; }
+			if (world[x+1][y].isAlive()) { numNeighbors++; }
```

Comment:
In the original version of the code, it checked against the other side of the world, e.g. if at 0,0 it also checked 0,ARRAY_SIZE.  This code doesn't do the same thing and it wasn't caught by tests.